_resources/port1.0/group/gpg_verify-1.0.tcl

```diff
@@ -45,7 +45,7 @@ options gpg_verify.gpg_homedir
 default gpg_verify.gpg_homedir {${workpath}/.gnupg}
 
 pre-checksum {
-     xinstall -o macports -m 0755 -d "[option gpg_verify.gpg_homedir]"
+     xinstall -o ${macportsuser} -m 0755 -d "[option gpg_verify.gpg_homedir]"
```

Comment:
I think this still won't work when not running as root.

Reply:
To be more specific, if `[geteuid] != 0` then you should not try to use xinstall's -o option.

Reply:
I’ll need some help creating a robust alternate for this then. I don’t know the situations when this stage isn’t run as root.

Subsequent code requires read/write access to `[option gpg_verify.gpg_homedir]`. That’s why the `-o ${macportsuser}` flag is in there.

Reply:
If you're not running as root then there is only one possible owner, so access shouldn't be a problem. This should be fine:
```Tcl
if {[geteuid] == 0} {
    xinstall -o ${macportsuser} -d [option gpg_verify.gpg_homedir]
} else {
    xinstall -d [option gpg_verify.gpg_homedir]
}
```

Or alternatively you could use chownAsRoot, but I don't really see a reason why that would be preferable.

Reply:
Thank you! Done.

Reply:
Also note that I implemented similar fixes in https://github.com/macports/macports-ports/pull/6049